test/integration/zap-mint.ts

```diff
@@ -0,0 +1,239 @@
+//move common variables to constants (like deadline, router)
+//add test for zap minting directly from ETH/WBTC (ETH/WBTC in path[0], path.length = 1)
+//add tests for all require statements
+//add test for reentrancy?
+describe("Zap mint", () => {
+  beforeEach(async () => {
+    await hre.deployments.fixture();
+    const deployer = await getDeployer(hre);
+    await setUpFunctionalSystemForTests(hre, 1);
+    const treasury = await getTreasurySigner(hre);
+    const bdx = await getBdx(hre);
+    const weth = await getWeth(hre);
+    const wbtc = await getWbtc(hre);
+    const allInputTokens = await getAllTokensAddresses(hre, [bdx.address, weth.address, wbtc.address]);
+    for (const token of allInputTokens) {
+      const erc20 = await getERC20(hre, token);
+      const decimals = await erc20.decimals();
+      erc20.connect(treasury).transfer(deployer.address, numberToBigNumberFixed(10, decimals));
+    }
+  });
+
+  it("Should zap mint for every possible non-singular path", async () => {
+    const deployer = await getDeployer(hre);
+    const bdx = await getBdx(hre);
+    const currentBlock = await hre.ethers.provider.getBlock("latest");
+    const weth = await getWeth(hre);
+    const router = await getUniswapRouter(hre);
+
+    const wbtc = await getWbtc(hre);
+    const collateralTokens = [weth.address, wbtc.address];
+    const excludedInputTokens = [bdx.address, weth.address, wbtc.address];
+    const allPossibleZapMintPaths = await generateAllPaths(hre, excludedInputTokens, collateralTokens);
+
+    const zapMintTestContract = (await hre.ethers.getContract("ZapMint", deployer)) as ZapMint;
+
+    const allBDStableCollateralPools = await getAllBDStablePools(hre);
+
+    for (const possibleZapMintPath of allPossibleZapMintPaths) {
+      const swapFrom = possibleZapMintPath[0];
+      const swapTo = possibleZapMintPath[possibleZapMintPath.length - 1];
+      const poolsWithProperCollateral: BdStablePool[] = [];
+      for (const pool of allBDStableCollateralPools) {
+        const collateralTokenAddress = await pool.collateral_token();
+        const stableTokenAddress = await pool.BDSTABLE();
+        if (collateralTokenAddress === swapTo && stableTokenAddress !== swapFrom) {
+          poolsWithProperCollateral.push(pool);
+        }
+      }
+
+      for (const poolWithProperCollateral of poolsWithProperCollateral) {
+        const stableExpectedFromMintAddress = await poolWithProperCollateral.BDSTABLE();
+        const stableExpectedFromMint = (await hre.ethers.getContractAt("IERC20", stableExpectedFromMintAddress)) as IERC20;
+        const tokenToApproveAndTranfer = await getERC20(hre, swapFrom);
+        const erc20 = await getERC20(hre, swapFrom);
+        const decimals = await erc20.decimals();
+        const amountIn = numberToBigNumberFixed(0.01, decimals);
+
+        const deployerStableBalanceBeforeMint = await stableExpectedFromMint.balanceOf(deployer.address);
```

Comment:
you should use a random user ( `getUser1()`) instead of deployer in tests where possible (e.g. for minting)

Reply:
done